src/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosure.java

```diff
@@ -496,14 +496,16 @@ private void findBackupFile(HttpMessage originalMessage) throws Exception {
 				sendAndReceive(requestmsg, false);
 				disclosedData = requestmsg.getResponseBody().getBytes();
 				int requestStatusCode = requestmsg.getResponseHeader().getStatusCode();
+				// If the response is empty it's probably not really a backup
+				boolean isEmptyResponse = requestmsg.getResponseHeader().getContentLength() == 0;
 
 				//just to complicate things.. I have a test case which for the random file, does NOT give a 404 (so gives404s == false) 
 				//but for a "Copy of" file, actually gives a 404 (for some unknown reason). We need to handle this case.
-				if ( 	( gives404s && requestStatusCode != HttpStatus.SC_NOT_FOUND) || 
+				if ( 	( gives404s && requestStatusCode != HttpStatus.SC_NOT_FOUND && !isEmptyResponse) || 
```

Comment:
The empty check could be done before both conditions. Worth extracting a method to make it easier to read, can be used here and in the parent check.